CONTRIBUTING.md

```diff
@@ -26,133 +26,170 @@ We specifically do not merge PRs **without prior issues** that:
+---
+
+## Writing code for OpenTofu
 
-This repository contains OpenTofu Core, which includes the command line interface, the main graph engine, and the documentation for them.
+Eager to get started on coding? Here's the short version:
 
-This document provides guidance on OpenTofu contribution recommended practices. It covers how to submit issues, how to get involved in the discussion, how to work on the code, and how to contribute code changes.
+1. Set up a Go development environment with Git.
+2. Pay attention to copyright: [please read the DCO](https://developercertificate.org/), write the code yourself, avoid copy/paste. Disable or limit your AI coding assistant.
+3. Run the tests with `go test` in the package you are working on.
+4. Build OpenTofu by running `go build ./cmd/tofu`.
+5. Update [the changelog](CHANGELOG.md).
+6. When you commit, use `git commit -s` to sign off your commits.
+
+---
 
-The easiest way to contribute is by [opening an issue](https://github.com/opentofu/opentofu/issues/new/choose)! Bug reports, broken compatibility reports, feature requests, old issue reposts, and well-prepared RFCs are all very welcome.
+### Setting up your development environment
 
-All major changes to OpenTofu Core go through the public RFC process, including those proposed by the core team. Thus, if you'd like to propose such a change, please prepare an RFC, so that the community can discuss the change and everybody has a chance to voice their opinion. You're also welcome to voice your own opinion on existing RFCs! You can find them by [going to the issues view and filtering by the rfc label](https://github.com/opentofu/opentofu/issues?q=is%3Aopen+is%3Aissue+label%3Arfc).
+You can develop OpenTofu on any platform you like. However, we recommend either a Linux (including WSL on Windows) or a MacOS build environment. You will need [Go](https://golang.org/) and [Git](https://git-scm.com/) installed, and we recommend an IDE to help you with code completion and code quality warnings. (We recommend installing the Go version documented in the [.go-version](.go-version) file.)
 
-Generally, we appreciate external contributions very much and would love to work with you on them. **However, please make sure to read the [Contributing a Code Change](#contributing-a-code-change) section prior to making a contribution.**
+Alternatively, if you use Visual Studio Code or Goland/IntelliJ and have Docker or Podman installed, you can also use a [devcontainer](.devcontainer.json). In Visual Studio Code, you can install the [Remote Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers), then reopen the project to get a prompt about activating the devcontainer. In Goland/Intellij, open the `.devcontainers.json` file and click the purple cube icon that appears next to the line numbers to activate the dev container. At this point you can proceed as if you were [building natively](#building-natively) on Linux.
```

Comment:
dev containers look interesting, I didn't try them, I'll check them. 

The dev containers are not directly related to how to debug the code, but I think the information in [DEBUGGING.md](https://github.com/opentofu/opentofu/blob/main/DEBUGGING.md) could be melded into this Contribution guide, or it can be mentioned there. I don't have a specific suggestion, feel free to skip this comment :)